### ISSUE TITLE
Hit result is shown empty even though it has data in lvl1 property of hit 

### ISSUE THREAD

**Issue by hakankaraduman** [algolia/docsearch#1167]
## Description

With the version 2, it was showing all results correctly with the default UI. After upgrading to v3, it shows some results empty. Even though it has lvl1 info in the found hit.

- DocSearch version: 3.0.0-alpha.41

See the images for the difference

<img width="661" alt="Screen Shot 2021-11-08 at 17 05 21" src="https://user-images.githubusercontent.com/8139393/140756998-ab895424-623e-4a2e-ace3-a1fcf22614de.png">
<img width="557" alt="Screen Shot 2021-11-08 at 17 08 03" src="https://user-images.githubusercontent.com/8139393/140757006-8be19323-3d91-4be9-92e4-a74f103e4c2b.png">

**Comment by shortcuts** [comment]
Hey, what is the concerned index? Are you using the Algolia Crawler or the legacy platform?

**Comment by hakankaraduman** [comment]
Yes, sometimes we initiate docsearch scraper on our own to speed up the process.

**Comment by shortcuts** [comment]
I'm not sure if it's because of that, but some `h2`s are missing `id`s on your website (e.g. https://docs.controlplane.com/reference/gvc `Overview`), so it might create a record with an empty anchor

**Comment by hakankaraduman** [comment]
I understand, thanks, we'll update the missing ids's and will try again.

**Comment by shortcuts** [comment]
No problem, feel free to close if it was the issue

**Comment by hakankaraduman** [comment]
Hi @shortcuts  , that wasn't the issue. Where can I find how you construct the "children" prop that's given to hitComponent prop. So I can use the same component but update the text as I like from the "hit" object I receive?

**Comment by shortcuts** [comment]
Hey @hakankaraduman, I can still see empty `id`s on production, was it tested on a dev env? I'd also like to try resolving this.

> Where can I find how you construct the "children" prop that's given to hitComponent prop. So I can use the same component but update the text as I like from the "hit" object I receive?

You can find it here: https://github.com/algolia/docsearch/blob/next/packages/docsearch-react/src/Results.tsx#L112, but I believe this issue should be solved at the indexing level

**Comment by hakankaraduman** [comment]
Hi @shortcuts ,  You are right, I deployed the fix to prod and triggered the manual crawler.

I also published a public test for you check out the v3 implementation. https://cpln-docs-test.web.app/

**Comment by hakankaraduman** [comment]
Empty results' hit object that's given to hitComponent property shows it crawled lvl1 data btw.

**Comment by shortcuts** [comment]
The issue could come to the fact that two pages with different URLs have the same `lvl0` (`h1`), so our frontend grouping does not find the right parent: https://github.com/algolia/docsearch/blob/next/packages/docsearch-react/src/DocSearchModal.tsx#L219

Trying with an other `lvl0` (`h2` for example) seems to solves the issue 🤔

**Comment by hakankaraduman** [comment]
Thank you for the update. Do you need anything else from us to help with your investigation?

**Comment by hakankaraduman** [comment]
Hi, happy new year @shortcuts and @bodinsamuel  ,

I was wondering if there is an investigation currently about this issue. Thanks

**Comment by enk21** [comment]
Hi @shortcuts and @bodinsamuel , were you able to investigate the issue? Anything else you need from our side? Thanks!
